mindmeld/active_learning/classifiers.py

```diff
@@ -0,0 +1,418 @@
+        test_queries: List[ProcessedQuery],
+        domain2id: Dict,
+        intent2id: Dict,
+    ):
+        """Fit and evaluate the domain classifier.
+        Args:
+            sampled_queries (List[ProcessedQuery]): List of Sampled Queries
+            unsampled_queries (List[ProcessedQuery]): List of Unsampled Queries
+            test_queries (List[ProcessedQuery]): List of Test Queries
+            domain2id (Dict): Dictionary mapping domains to IDs
+            intent2id (Dict): Dictionary mapping intents to IDs
+
+        Returns:
+            ic_queries_prob_vectors (List[List]): List of probability distributions
+                for unsampled queries
+            ic_eval_test_dict (Dict): Dictionary mapping a domain (str) to the
+                associated ic_eval_test object
+        """
+        ic_eval_test_dict = {}
+        unsampled_idx_preds_pairs = []
+        for domain in domain2id:
+            # Filter Queries
+            _, filtered_sampled_queries = DataBucket.filter_queries(
+                sampled_queries, domain
+            )
+            (
+                filtered_unsampled_queries_indices,
+                filtered_unsampled_queries,
+            ) = DataBucket.filter_queries(unsampled_queries, domain)
+            _, filtered_test_queries = DataBucket.filter_queries(test_queries, domain)
+            # Train
+            ic = self.nlp.domains[domain].intent_classifier
+            ic.fit(queries=filtered_sampled_queries)
+            # Evaluate Test Queries
+            ic_eval_test = ic.evaluate(queries=filtered_test_queries)
+            if not ic_eval_test:
+                unsampled_idx_preds_pairs.extend(
+                    (i, self._get_zero_prob_vector())
+                    for i in filtered_unsampled_queries_indices
+                )
+                continue
+            ic_eval_test_dict[domain] = ic_eval_test
+            # Get Probability Vectors
+            ic_queries_prob_vectors = MindMeldALClassifier._get_probs(
+                classifier=ic,
+                queries=filtered_unsampled_queries,
+                nlp_component_to_id=intent2id[domain],
+            )
+            intents = [
+                f"{domain}|{intent}"
+                for intent in ic_eval_test.get_stats()["class_labels"]
+            ]
+            padded_ic_queries_prob_vectors = self._pad_intent_probs(
+                ic_queries_prob_vectors, intents
+            )
+            for i in range(len(filtered_unsampled_queries)):
+                unsampled_idx_preds_pairs.append(
+                    (
+                        filtered_unsampled_queries_indices[i],
+                        padded_ic_queries_prob_vectors[i],
+                    )
+                )
+        unsampled_idx_preds_pairs.sort(key=lambda x: x[0])
+        padded_ic_queries_prob_vectors = [x[1] for x in unsampled_idx_preds_pairs]
+        return padded_ic_queries_prob_vectors, ic_eval_test_dict
+
+    def _get_zero_prob_vector(self):
```

Comment:
Nit: Why have a special function only for length = number of intents? 

Can we either make it a general function that takes in length as a parameter and return a zero prob vector or remove this and replace line 362 since that is the only place it's used?

Reply:
Replaced with `(i, np.zeros(len(self.intent2idx)))`